docs/sql-data-sources-text.md

```diff
@@ -0,0 +1,40 @@
+---
+layout: global
+title: TEXT Files
+displayTitle: TEXT Files
+license: |
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+ 
+     http://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+Spark SQL provides `spark.read().text("file_name")` to read a file or directory of files in TEXT format into Spark DataFrame, and `dataframe.write().text("path")` to write to a TEXT file. Function `option()` can be used to customize the behavior of reading or writing, such as controlling behavior of the wholetext, line separator, compression, and so on.
+
+<!--TODO: add `option()` document reference-->
```

Comment:
Shall we add the JIRA id here? TODO(SPARK-XXX): ...

Reply:
TEXT doesn't need to be capitalized. In fact it's not really a format. Just 'a text file or directory of text files'?
into Spark -> into a Spark
"The option() function can ..."
If we're not explaining "wholetext" here, I'd omit it, as it's not clear what that means.

Reply:
So, while I appreciate having any docs, none of this really explains even the most basic info about reading/writing text. For example, reading a text file gives you what in a DataFrame?

Reply:
I see, let me improve the sentence.
Thanks, @HyukjinKwon and @srowen